### PR TITLE
[Hub Generated] [Microsoft.Cache] 2022-05-01 API: Add "Latest" option to "redisVersion" property, async linked server, managed identity changes 

### PR DESCRIPTION
**Pull request by JeffreyCA**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/JeffreyCA/azure-rest-api-specs/dev-redis-Microsoft.Cache-2022-05-01...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
    July 2022

  4. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
    July 2022
  6. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/redis/resource-manager/Microsoft.Cache/stable/2020-12-01/redis.json [hunk 1]
```diff
@@ -1884,7 +1884,7 @@
         },
         "redisVersion": {
           "type": "string",
-          "description": "Redis version. Only major version will be used in PUT/PATCH request with current valid values: (4, 6)"
+          "description": "Redis version. This can be in the form 'major[.minor[.build]]' (only 'major' is required) or the value 'latest' which refers to the latest stable Redis version that is available. Only the major and minor version are used in a PUT/PATCH request. Supported versions: 4.0, 6.0."
```
[the remark this answers is not in the document]
Comment (JeffreyCA) [reply specification/redis/resource-manager/Microsoft.Cache/stable/2020-12-01/redis.json]: No, the service always returns the full version with major, minor, and build components (e.g. 6.0.14).

File: specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json [hunk 2]
```diff
@@ -0,0 +1,2815 @@
...
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Redis"
+        ],
+        "operationId": "Redis_Update",
+        "x-ms-long-running-operation": true,
```
Comment (rkmanda) [inline specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json:252]: >x-ms-long-running-operation": true, [](http://example.com/codeflow?start=9&length=35)

I only see the patch operation getting converted to long running in this PR. when I do a diff between iteration 10 and 1. Can you ping me on teams so we can work thru the actual diffs? [alias - rmanda]

Comment (rkmanda) [inline specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json:252]: >ms [](http://example.com/codeflow?start=11&length=2)

converting a sync API to async is considered a cross version breaking change. Why is this being done? Can you achieve the update using a long running POST instead?

Comment (JeffreyCA) [reply specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json]: We've already gotten a signoff for these async-related breaking changes. They were part of a [previous PR](https://github.com/Azure/azure-rest-api-specs/pull/17320), but our team decided to include them in a new API version instead. @chantalolieman can provide more context if needed.

Comment (JeffreyCA) [reply specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json]: The examples and specs have been updated since the last review

**Comment by JeffreyCA** [comment]
Hi @weidongxu-microsoft can you please review this?

File: specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json [hunk 3]
```diff
@@ -282,6 +293,12 @@
               "$ref": "#/definitions/RedisResource"
             }
           },
+          "202": {
+            "description": "The existing redis cache was successfully updated. Check Azure-AsyncOperation headers to poll detailed status.",
+            "schema": {
+              "$ref": "#/definitions/RedisResource"
+            }
+          },
```
Comment (weidongxu-microsoft) [inline specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json:301]: Is this PATCH long running operation?

**Comment by weidongxu-microsoft** [comment]
@msyyc @tadelesh @qiaozha @XiaofeiCao @fengzhou-msft for awareness to breaking changes (Jeffrey already approved).

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by msyyc** [comment]
approved for python breaking change

File: specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json (hunk 3 above)
Comment (JeffreyCA) [reply specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json:302]: Yes, I pushed some updates to add `x-ms-long-running-operation` and addressed the warnings

**Comment by austintolani** [comment]
Hello @weidongxu-microsoft  and @msyyc. I am on the Azure Cache for Redis Team and am taking over this work from @JeffreyCA. The RP manifest rollout has completed for all regions and we are ready to have this PR merged. Yesterday, I removed the `DoNotMerge` tag from the PR. Is there anything else that I need to do to merge the PR?

**Pull request merged**